app/security_system.py

```diff
@@ -66,13 +67,43 @@ def recognize_face(self, frame):
 
         # TODO 3
         # Perform face detection
+        faces = face_cascade.detectMultiScale(gray_frame, scaleFactor = 1.2, minNeighbors = 5, minSize = (30,30))
         # Iterate through each detected face
-        # Extract the detected face region
-        # Perform facial recognition on the detected face
-        # Draw bounding box around the detected face
-        # Annotate the frame with the recognized user's name and authorization status
-        # Log access attempt
-        # Your code goes here
+        for face in faces:
+            # Extract the detected face region
+            x,y,w,h = face
+            X,Y = x+w,y+h
+            image = gray_frame[y:Y, x:X]
+            # Perform facial recognition on the detected face
+            label, confidence = self.face_recognizer.predict(image)
+            person_name = self.get_person_name(label,confidence)
+            authorization_status = self.authorized_persons[person_name]
+            if(authorization_status==True):
+                color = (0,200,100)
+                is_authorized = "YES"
+            else:
+                color = (0,35,200)
+                is_authorized = "NO"
```

Comment:
Try to use the already implemented function `is_person_authorized` to get the authorization status. We're doing this because currently we have a simple file to get the authorization status but later we might wanna change that, in that case it makes more sense to implement it as a separate function.

You might need to change the function body of `is_person_authorized` as it may not be correctly implemented.